substrate/client/src/client.rs

```diff
@@ -369,8 +378,11 @@ impl<B, E, Block> Client<B, E, Block> where
 		let unchecked: bft::UncheckedJustification<_> = justification.uncheck().into();
 		transaction.set_block_data(header.clone(), body, Some(unchecked.into()), is_new_best)?;
 		transaction.update_authorities(authorities);
-		if let Some(storage_update) = storage_update {
+		if let Some((storage_update, changes)) = storage_update {
 			transaction.update_storage(storage_update)?;
+			// TODO [ToDr] How to handle re-orgs? Should we re-emit all storage changes?
+			self.storage_notifications.lock()
```

Comment:
Are you sure that it should be called **before** transaction is committed?

Reply:
Good point. Moved the notification after `commit` and also guarded by the same `if` as block import notification.